src/hera/_cli/main.py

```diff
@@ -0,0 +1,25 @@
+"""The main entrypoint for hera CLI."""
+from __future__ import annotations
+
+from hera._cli import generate
+
+
+def main():
+    try:
+        import typer
+    except ImportError:
+        raise ImportError("Use of the `hera` CLI tool requires installing the 'cli' extra, `pip install hera[cli]`.")
+
+    app = typer.Typer()
+
+    generate_app = typer.Typer()
```

Comment:
this is admittedly kind of weird. Typer, because of its click origins, (imo) makes testing more challenging because it takes over your function when using their normal decoration pattern. I did this so that `generate_yaml` could be directly called in tests for the majority of the testing, rather than needing to either use their test runner or define extra functions to enable the same thing.

I can certainly undo this and use the more natural decorator style, although i anticipate that blowing up the numbers of boilerplate files and functions required to achieve the same thing